// karma-coverage/Gruntfile.js
module.exports = function(grunt) {
  grunt.initConfig({
    pkgFile: 'package.json',
    files: {
      adapter: ['src/adapter.js']
    },
    build: {
      adapter: '<%= files.adapter %>'
    },
    jshint: {
      adapter: {
        files: {
          src: '<%= files.adapter %>'
        },
        options: {
          browser: true,
          es5: true,
          strict: false,
          undef: false,
          camelcase: false
        }
      },
      options: {
        quotmark: 'single',
        camelcase: true,
        strict: true,
        trailing: true,
        curly: true,
        eqeqeq: true,
        immed: true,
        latedef: true,
        newcap: true,
        noarg: true,
        sub: true,
        undef: true,
        boss: true,
        globals: {}
      }
    },
    karma: {
      adapter: {
        configFile: 'karma.conf.js',
        autoWatch: false,
        singleRun: true,
        reporters: ['dots']
      }
    }
  });
  grunt.loadTasks('tasks');
  grunt.loadNpmTasks('grunt-contrib-jshint');
  grunt.loadNpmTasks('grunt-karma');
  grunt.registerTask('default', ['build', 'jshint', 'test']);
  return grunt.registerTask('test', ['karma']);
};
